### PR TITLE
[WebProfilerBundle] Make WDT follow ajax requests if header set

### PR DESCRIPTION
**Pull request by jeffreymb**
Replaces #22509. I accidentally closed that PR and couldn't get GitHub to recognize when I added more commits to the branch in my fork.

| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes (There are no tests that I could find)
| Fixed tickets | #15456
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

When the header of an ajax response contains the `Symfony-Debug-Toolbar-Replace` header with a value of '1' it will automatically update the toolbar with the toolbar with the debug info from the ajax request.

The bulk of the code in the `loadToolbar` function in the `base_js.html.twig` file is moved from the `toolbar_js.html.twig` file and slightly refactored to use the internal functions instead of making external calls. I take no credit/blame for this code. 😉 

If this could make it into 4.1 there are multiple people that would be very happy!

### CODE REVIEW

**Comment by gharlan** [comment]
@jeffreymb tested it again and found a problem. When the ajax page triggers immediately another ajax call (without the `-Replace` header), the ajax block in toolbar doesn't stop blinking.

It is reproducible for me with these controllers:

```php
    /**
     * @Route("/")
     */
    public function indexAction(): Response
    {
        return new Response(<<<'HTML'
<html>
<head>
</head>
<body>
    <a href="/foo">Load /foo</a>
    <main></main>
    
    <script src="https://code.jquery.com/jquery-3.3.1.min.js"></script>
    <script>
        $('a').click(function (event) {
            event.preventDefault();
            $('main').load('/foo');
        });
    </script>
</body>
</html>
HTML
);
    }

    /**
     * @Route("/foo")
     */
    public function fooAction(): Response
    {
        $response = new Response('
            foo
            <script>$.get(\'/bar\');</script>
        ');
        $response->headers->set('Symfony-Debug-Toolbar-Replace', 1);

        return $response;
    }

    /**
     * @Route("/bar")
     */
    public function barAction(): Response
    {
        return new Response('', 204);
    }
```

Load index page and click on "Load /foo". After that the ajax block is blinking all the time here.

**Comment by jeffreymb** [comment]
@gharlan This PR doesn't (at this point) touch the code that is relevant to the issue you've discovered. Is this for sure in scope for this PR?

**Comment by gharlan** [comment]
Without this PR I don't have this issue.

**Comment by jeffreymb** [comment]
@gharlan fixed

**Comment by gharlan** [comment]
Thanks, it works great! ❤️ 

In `base_js.html.twig` I found this comment:

> This file is partially duplicated in TwigBundle/Resources/views/base_js.html.twig. If you
   make any change in this file, verify the same change is needed in the other file.

So maybe you have to copy your changes to this file? I'm not sure.

**Comment by jeffreymb** [comment]
I've double checked that none of the changes in this PR are relevant to `TwigBundle/Resources/views/base_js.html.twig`.

**Comment by jeffreymb** [comment]
Any chance this can make 4.1? I'm not sure what the policy is, but this PR was created before the feature freeze, and actually most of the work done before the 4.0 feature freeze!

**Review by nicolas-grekas — APPROVED** [review]
Would be great to have a doc PR also.

**Comment by jeffreymb** [comment]
There is a docs issue: https://github.com/symfony/symfony-docs/issues/8409

**Comment by nicolas-grekas** [comment]
ping @symfony/deciders

File: src/Symfony/Bundle/WebProfilerBundle/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ CHANGELOG
 -----
 
  * added information about orphaned events
+ * added ability to auto-refresh the toolbar if a `Symfony-Debug-Toolbar-Replace` header
+    is set and has a value of '1'
```
Comment (fabpot) [inline src/Symfony/Bundle/WebProfilerBundle/CHANGELOG.md]: extra whitespace at the beginning of the line

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/WebProfilerBundle/CHANGELOG.md (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Bundle/WebProfilerBundle/CHANGELOG.md]: here is what I propose here:
made the toolbar auto-update with info from ajax reponses when they set the `Symfony-Debug-Toolbar-Replace` header to `1`

I would have done it myself, but you disallowed me from pushing on your fork :)

**Review by javiereguiluz — APPROVED** [review]

File: src/Symfony/Bundle/WebProfilerBundle/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,8 @@ CHANGELOG
 -----
 
  * added information about orphaned events
+ * made the toolbar auto-update with info from ajax reponses when they set the 
+   `Symfony-Debug-Toolbar-Replace header to `1`
```
Comment (sroze) [inline src/Symfony/Bundle/WebProfilerBundle/CHANGELOG.md]: Missing \` after `Symfony-Debug-Toolbar-Replace`

**Comment by jeffreymb** [comment]
I created a doc PR earlier today.

**Comment by nicolas-grekas** [comment]
Thank you @jeffreymb.

**Pull request merged**